### ISSUE TITLE
How to migrate nginx to contour feature request

### ISSUE THREAD

**Issue by dkolovrat** [projectcontour/contour#3937]
Can we make a scripts or solution how to migrate from nginx scripts to contour,  we can do a “path rewriting” and “header rewriting” discriminating by a prefix, but we not have find anything how to set an “if” with a request method:
example: if ($request_method = 'OPTIONS')
If we using on another  environment we use ngnix, and we have a lot of rules in the annotations block, below an extract. Have you a suggestion to migrate from this to contour?

>      location /tool {
            if ($request_method = 'OPTIONS')
             {
             add_header 'Access-Control-Allow-Origin' '*';
             add_header 'Access-Control-Allow-Credentials' 'true';
             add_header 'Access-Control-Allow-Headers' 'Authorization,Accept,Origin,DNT,X-CustomHeader,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Content-Range,Range,Csrf-Token,X-Crsf-Token';
             add_header 'Access-Control-Allow-Methods' 'GET,POST,OPTIONS,PUT,DELETE,PATCH';
             add_header 'Access-Control-Max-Age' 1728000;
             add_header 'Content-Type' 'text/plain charset=UTF-8';
             add_header 'Content-Length' 0;
             return 204;
            }
             rewrite              /tool/(.*) /api/$1 break;
             proxy_set_header X-Real-IP $remote_addr;
             proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
             proxy_set_header X-NginX-Proxy true;
             proxy_set_header   Upgrade $http_upgrade;
             proxy_set_header   Connection "upgrade";
             proxy_pass           $tool_endpoint;
             #proxy_ssl_session_reuse off;
             #proxy_set_header Host $http_host;
             #proxy_cache_bypass $http_upgrade;
             proxy_redirect off;

**Comment by youngnick** [comment]
Hi @dkolovrat, sorry about the delay in getting back to you.

We support more advanced routing features like this using our HTTPProxy CRD.

Header rewriting and adding are first-class features there, you can see https://projectcontour.io/docs/v1.18.0/config/request-rewriting/#header-rewriting for more information on how you can do that.

Additionally, for request method routing, you can use the `:method` pseudo-header, as documented in #2679.

Basically, you'll end up with a HTTPProxy with two `routes:`, one that matches the `:method` header, and one that doesn't, and then use the rewrite support we already have to duplicate what you have in your script, I think.
